### PR TITLE
Cls2 340 add objectives and blockers front end

### PR DESCRIPTION
**Pull request by PaulWheatcroft**
## Description of change

Allow users to add multiple objectives to a single company.

## Test instructions

*requires latest API source files if running locally
Pull this branch. Navigate to http://localhost:3000/companies/<company_id>/account-management/objective/create
Create and submit an objective.

## Screenshots

Screenshot 2023-08-08 at 11 06 07

Screenshot 2023-08-08 at 11 05 04

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'

- [x] Has the branch been rebased to main?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

**Review by backslashbaker — APPROVED** [review]

File: src/client/modules/Companies/AccountManagement/tasks.js [hunk 1]
```diff
@@ -1,7 +1,19 @@
...
+import { transformRadioOptionToBool } from '../../Investments/Projects/transformers'
 
 export const saveStrategy = ({ strategy, companyId }) => {
   const request = apiProxyAxios.patch
   const endpoint = `/v4/company/${companyId}`
   return request(endpoint, { strategy: strategy })
 }
+
+export const saveObjective = ({ values, companyId }) => {
+  values.target_date = transformValueForAPI(values.target_date)
```
Comment (chopkinsmade) [inline src/client/modules/Companies/AccountManagement/tasks.js:12]: We usually have a separate function to do these transformations, take a look at `src/client/modules/ExportPipeline/transformers.js` and `src/client/modules/ExportPipeline/ExportForm/tasks.js` for some examples. There are tests in `src/client/modules/ExportPipeline/__test__/transformers.test.js`

File: src/client/modules/Companies/AccountManagement/Objective.jsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+              submitButtonLabel="Save objective"
+              cancelButtonLabel="Back"
+            >
+              {() => (
+                <>
+                  <FieldInput
+                    type="text"
+                    name="subject"
+                    label="Objective subject"
+                    required="Enter a objective subject"
```
Comment (chopkinsmade) [inline src/client/modules/Companies/AccountManagement/Objective.jsx:91]: Should be "Enter a**n** objective subject"

File: src/client/modules/Companies/AccountManagement/Objective.jsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  FieldRadios,
+} from '../../../components'
+import urls from '../../../../lib/urls'
+import { CompanyResource } from '../../../components/Resource'
+import { buildCompanyBreadcrumbs } from '../utils'
+import { OPTIONS_YES_NO, OPTION_YES } from '../../../../apps/constants'
+
+const Objective = () => {
+  const { companyId } = useParams()
+  const OBJECTIVE_PERCENTAGE = {
```
Comment (chopkinsmade) [inline src/client/modules/Companies/AccountManagement/Objective.jsx:22]: You can switch this to just an array of label / value objects, and changing `options={Object.values(OBJECTIVE_PERCENTAGE)}` to `options={OBJECTIVE_PERCENTAGE}`

File: src/client/modules/Companies/AccountManagement/Objective.jsx (hunk 2 above)
Comment (PaulWheatcroft) [reply src/client/modules/Companies/AccountManagement/Objective.jsx:91]: Done

File: test/functional/cypress/specs/companies/account-management-add-objective-spec.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+      cy.get('[data-test="target_date-day"]').click().type('32')
+      cy.get('[data-test="target_date-month"]').click().type('18')
+      cy.get('[data-test="target_date-year"]').click().type('030')
+      cy.get('[data-test="has-blocker-no"]').click()
+      cy.get('[data-test="progress-50"]').click()
+      cy.intercept('POST', `/api-proxy/v4/company/${companyId}/objective`, {
+        results: noBlockersObjective,
+      })
+      cy.get('[data-test="submit-button"]').click()
+      cy.get('[data-test="summary-form-errors"]')
```
[the remark this answers is not in the document]
Comment (PaulWheatcroft) [reply test/functional/cypress/specs/companies/account-management-add-objective-spec.js:106]: The target date field doesn't work with assertFieldError as it is a group of fields and not a single field

File: src/client/modules/Companies/AccountManagement/tasks.js (hunk 1 above)
Comment (PaulWheatcroft) [reply src/client/modules/Companies/AccountManagement/tasks.js]: Transformer created + associated test

File: test/functional/cypress/specs/companies/account-management-add-objective-spec.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+      })
+
+      it('should contain all required form inputs', () => {
+        fill('[data-test="subject-input"]', noBlockersObjective.subject)
+        fill('[data-test="target_date-day"]', '10')
+        fill('[data-test="target_date-month"]', '10')
+        fill('[data-test="target_date-year"]', '2030')
+        cy.get('[data-test="has-blocker-no"]').click()
+        cy.get('[data-test="progress-50"]').click()
+        cy.intercept('POST', `/api-proxy/v4/company/${companyId}/objective`, {
```
Comment (chopkinsmade) [inline test/functional/cypress/specs/companies/account-management-add-objective-spec.js:55]: Can you add this intercept to the before() statement, and put it before the cy.visit() with the other one. I've seen flaky tests with the intercept calls not setup before visiting a page

File: test/functional/cypress/specs/companies/account-management-add-objective-spec.js [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+      cy.visit(urls.companies.accountManagement.objectives.create(companyId))
+    })
+    it('should highlight an error with the date input', () => {
+      fill('[data-test="subject-input"]', noBlockersObjective.subject)
+      fill('[data-test="target_date-day"]', '32')
+      fill('[data-test="target_date-month"]', '18')
+      fill('[data-test="target_date-year"]', '030')
+      cy.get('[data-test="has-blocker-no"]').click()
+      cy.get('[data-test="progress-50"]').click()
+      cy.intercept('POST', `/api-proxy/v4/company/${companyId}/objective`, {
```
Comment (chopkinsmade) [inline test/functional/cypress/specs/companies/account-management-add-objective-spec.js:93]: Don't need intercept on a validation failure test, as the form doesn't get as far as the POST

File: test/functional/cypress/specs/companies/account-management-add-objective-spec.js [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+      fill('[data-test="target_date-year"]', '2020')
+      cy.get('[data-test="has-blocker-no"]').click()
+      cy.get('[data-test="field-blocker_description"]').should('not.exist')
+      cy.get('[data-test="has-blocker-yes"]').click()
+      cy.get('[data-test="field-blocker_description"]')
+        .should('exist')
+        .click()
+        .type(noBlockersObjective.blocker_description)
+      cy.get('[data-test="progress-25"]').click()
+      cy.intercept('POST', `/api-proxy/v4/company/${companyId}/objective`, {
```
Comment (chopkinsmade) [inline test/functional/cypress/specs/companies/account-management-add-objective-spec.js:122]: Same as above, can you add this intercept to the before() statement,

File: test/functional/cypress/specs/companies/account-management-add-objective-spec.js (hunk 6 above)
Comment (PaulWheatcroft) [reply test/functional/cypress/specs/companies/account-management-add-objective-spec.js]: Done

File: test/functional/cypress/specs/companies/account-management-add-objective-spec.js (hunk 5 above)
Comment (PaulWheatcroft) [reply test/functional/cypress/specs/companies/account-management-add-objective-spec.js]: Done

**Review by chopkinsmade — APPROVED** [review]

**Pull request merged**